3737/1/parser.rb

```diff
@@ -0,0 +1,98 @@
+require 'nokogiri'
+require 'open-uri'
+require 'xpath'
+
+# class for parsing info about gem from github page
+class Parser
+  def initialize
+    @arr = {}
+  end
+
+  def repository(gname)
+    link = search_repo(gname)
+    return unless link
+    gethash(link)
+  end
+
+  private
+
+  def search_repo(gname)
+    url = "http://rubygems.org/gems/#{gname}"
+    check_page('code', url) || check_page('home', url)
+  end
+
+  def check_page(linkid, url)
```

Comment:
what actually does this method? it finds the link to github repo on rubygems page. it should be reflected in methods names